### PR TITLE
update CI to build on PRs

### PR DESCRIPTION
**Pull request by VijayanB**
Signed-off-by: Vijayan Balasubramanian <balasvij@amazon.com>
### Description
Update build settings to get dependencies from local. In the meantime, update CI to
build OpenSearch and publish to local maven.
  
### Check List
- [ ] New functionality includes testing.
  - [X] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — COMMENTED** [review]
This looks good. You got a duplicate test workflow that you'll want to merge/delete to get this PR to 🍏  (probably copy the codecov parts into the CI one).

**Comment by dblock** [comment]
Also open an issue and link here for anything you're removing temporarily that needs to be fixed (e.g. integ tests).

**Comment by VijayanB** [comment]
Commented integration tests. Will track it here #6

File: build.gradle [hunk 1]
```diff
@@ -110,7 +91,7 @@ if (!usingRemoteCluster) {
 }
 
 ext {
-    opendistroVersion = '1.15.0'
+    opendistroVersion = '1.0.0'
```
Comment (jmazanec15) [inline build.gradle:94]: Why not update opendistroversion here to opensearch?

File: build.gradle [hunk 2]
```diff
@@ -151,7 +132,7 @@ task release(type: Copy, group: 'build') {
     from(zipTree(project.tasks.bundlePlugin.outputs.files.getSingleFile()))
     into "build/plugins/opendistro-knn"
```
Comment (jmazanec15) [inline build.gradle:133]: Why not change this to opensearch? Is that outside scope of PR?

File: build.gradle (hunk 1 above)
Comment (VijayanB) [reply build.gradle:94]: Will target complete renaming variable/namespace in next phase.

File: build.gradle (hunk 2 above)
Comment (VijayanB) [reply build.gradle:133]: Thats correct. I tried to rename only obvious quoted strings that will not have impact on code.

**Review by jmazanec15 — APPROVED** [review]

**Review by vamshin — APPROVED** [review]
LGTM! Thanks

**Pull request merged**